03-MVC-AngularJS/js/script.js

```diff
@@ -0,0 +1,86 @@
+var myApp = angular.module('myApp', ['ngRoute']);
+
+myApp.config(function($routeProvider) {
+    $routeProvider
+    .when("/:id", {
+        template : (`<h1>Title: {{selected.title}}</h1><br>
+        <p>Duration: {{selected.duration}}</p>
+        <p>Year: {{selected.year}}</p>
+        <p>Synopsis: {{selected.synopsis}}</p>`)
+    })
+});
+
+myApp.service('storage',[function() {
+
+    this.save  = function (moviesUnsaved) {
+        localStorage.setItem('movies', angular.toJson(moviesUnsaved));
+    }
+
+    this.get  = function() {
+        let movies = JSON.parse(localStorage.getItem('movies'));
+
+        if (movies === null) {
+            movies = [];
+        }
+        return movies;
+    }
+}]);
+
+myApp.controller('mainController', ['$scope', 'storage', function ($scope, storage) {
+    let moviesUpdate;
+
+    $scope.movies = {};
+    $scope.selected = {};
+    $scope.storage = storage;
+    $scope.movies.details = $scope.storage.get();
+    moviesUpdate = $scope.movies.details;
+    $scope.isDisabled = true;
+    $scope.showDetails = false;
+
+    $scope.$watch('movies.details', function(moviesUpdate) {$scope.storage.save(moviesUpdate);}, true);
```

Comment:
You can provide storage as a parameter to the callback, this way you avoid assigning it to the $scope.